tests/workspace_lint_config.rs

```diff
@@ -0,0 +1,26 @@
+use assert_cmd::Command;
+
+#[test]
+fn test_workspace_config() {
+    // TODO: we are only checking for required semver version overrides
+    // right now because that is all that is currently implemented.
+    let mut cmd = Command::cargo_bin("cargo-semver-checks")
+        .expect("cargo semver-checks command should exist");
+
+    cmd.current_dir("test_crates/manifest_tests/workspace_overrides")
+        .args([
+            "semver-checks",
+            "--baseline-root",
+            "old", // should be the workspace root, not package root
+            "--manifest-path",
+            "new/pkg/Cargo.toml",
```

Comment:
Possible other test: pass `--manifest-path new/Cargo.toml` -- the workspace manifest instead of the individual crate manifest -- and assert that the behavior is the same.

We want to make sure that crates are scanned with _their crate's_ settings, even when the scan is triggered at workspace level.

This is fine to add in a subsequent PR if that's easier.